stl/inc/xstring

```diff
@@ -1209,6 +1209,13 @@ public:
         "Bad char_traits for basic_string_view; "
         "N4659 24.4.2 [string.view.template]/1 \"the type traits::char_type shall name the same type as charT.\"");
 
+    static_assert(!is_array_v<typename _Traits::char_type>,
+        "Bad char_traits for basic_string_view; Character type of basic_string_view must not be an array");
+
+    static_assert(is_trivial_v<typename _Traits::char_type> && is_standard_layout_v<typename _Traits::char_type>,
+        "Bad char_traits for basic_string_view; LWG-3034: implementations of basic_string and basic_string_view that "
+        "expect character types to be both trivial and standard layout.");

```

Comment:
```suggestion
    static_assert(!is_array_v<_Elem> && is_trivial_v<_Elem> && is_standard_layout_v<_Elem>,
        "The character type of basic_string_view must be a non-array trivial standard-layout type. See N4861 [strings.general]/1.");

```

I don't think we should direct users to documents that aren't the standard when that can be avoided. Also note that we just asserted that _Elem is char_type so it can be shorter referring to that.

Ditto below.